### ISSUE TITLE
emacs exits immediately without any output

### ISSUE THREAD

**Issue by lianhongHou** [babun/babun#694]
As title, emacs exits immediately when it is opened in babun shell, and it leaves nothing on shell.
It prompts "The procedure entry point __locale_ctype_ptr could not be located in the dynamic link library cygwin1.dll." when .babun/cygwin/bin/emacs-nox.exe is double-clicked.

The command "pact install emacs" is used to install emacs, and it seems emacs25.1 is installed.

**Comment by JerzySpendel** [comment]
I have the same problem with cmake and emacs.

**Comment by zhizunbao84** [comment]
i use the command "pact install emacs" to install emacs, and it installs emacs25.1
but i cannot run emacs!
![qq 20160922164716](https://cloud.githubusercontent.com/assets/2263408/18741981/818b81f6-80e4-11e6-8be1-ee6cb5b20e22.png)

**Comment by Explorare** [comment]
Same problem with VIM 8 and GunPG.

**Comment by rongyi** [comment]
I have the same problem
babum version 1.2

**Comment by svbeon** [comment]
same problem with irssi 0.8.20-1 after updating from irssi 0.8.19-1

**Comment by svbeon** [comment]
breaks it even more...
`C:/home/.babun/cygwin/bin/irssi.exe: error while loading shared libraries: ?: cannot open shared object file: No such file or directory`

**Comment by Explorare** [comment]
@svbeon Please make sure you installed 32-bit version of Cygwin.

**Comment by svbeon** [comment]
@Explorare i did

**Comment by Explorare** [comment]
@svbeon Umm... I've tried install emacs in my babun, it works. (But I don't know how to exit it lol. I think it should be `q!`)
I've updated Cygwin yesterday
`CYGWIN_NT-6.3-WOW Explorare-VAIO 2.6.0(0.304/5/3) 2016-08-31 14:27 i686 Cygwin`

**Comment by svbeon** [comment]
@Explorare do you get irssi running? Because that's the package I have problems with

**Comment by Explorare** [comment]
@svbeon Yep, I could. So did u tried remove babun completely and reinstall again?

**Comment by svbeon** [comment]
@Explorare multiple times

Den tors 6 okt. 2016 10:36Explorare <notifications@github.com> skrev:

**Comment by Explorare** [comment]
@svbeon Umm... Sry but I'm noob at debug. ( ´_ゝ`) May the force be with you.

**Comment by kielstrand** [comment]
did you try using the update batch in the babun directory? this updates cygwin and all depend...

for me this fixed all the issues regarding 127

cheers

**Comment by svbeon** [comment]
Yes, sadly it didn't solve it

Den tors 13 okt. 2016 16:32kielstrand <notifications@github.com> skrev:

**Comment by kielstrand** [comment]
ok I tried the following:
Clean Babun install --> installing emacs using pact.
--> Emacs is broken.
Running update.bat fixes this, for all the broken packages installed with pact. I am running windows 10

**Comment by mkub1ak** [comment]
Invoking $HOME\.babun\update.bat on Windows 10 solved the issue for me.
Thanks @kielstrand !

**Comment by wastrelman** [comment]
run ../.babun/update.bat on Win 7, and solved this issue for me too, thanks a lot ~

**Comment by GabLeRoux** [comment]
```
{ ~ }  » uname -a                                                                                                                                                                                                                        ~
CYGWIN_NT-6.1 IE8Win7 1.7.35(0.287/5/3) 2015-03-04 12:07 i686 Cygwin
```

`IEUser` is my username, ran
```
open  /c/Users/IEUser/.babun/update.bat
```
Then ran 
```
pact remove emacs
pact install emacs
```

Still getting 127 and no output.

Does the exact same thing with `tmux`.

**Comment by shinmai** [comment]
>     { ~ }  » uname -a                                                                                                                                                                                                                    
>     CYGWIN_NT-6.1 IE8Win7 1.7.35(0.287/5/3) 2015-03-04 12:07 i686 Cygwin
> > 
> Did a fresh install.
> 
> 
> IEUser is my username, ran
> 
>     open  /c/Users/IEUser/.babun/update.bat
> 
> 
I did this...
> Then ran
> 
>     pact remove emacs
>     pact install emacs
...but didn't do this, and tmux and emacs are both working fine now.

**Comment by Echsecutor** [comment]
I can confirm the issue. Fresh install and `pact install emacs` in babun: running `emacs` will result in 127. `emacs-nox` is not found. -> `emacs -nw` doesn't work.

```
 » babun update                                                                                                                                                                 ~/cygwin-home/data_analyzer
Executing babun update
  upstream  [release]
  installed [1.2.0]
  newest    [1.2.0]
Babun is up to date
Checking Cygwin version:
  installed [1.7.35]
  newest    [1.7.35]
Cygwin is up to date

```
Does not do anything, however running the update.bat from Windows directly triggers lots of cygwin upgrade activity.  Now `pact install emacs` yields a working `emacs`.

**Comment by Echsecutor** [comment]
After running the update.bat, make does not work any more (before it did). A dll is gone: `cygcheck make`
reports `cygcheck: track_down: could not find cygcrypt-0.dll`. This can now be fixed by `pact update libcryptox`. But hey... that's not much less painful then using cygwin directly. ;)

**Comment by Eduard0J0** [comment]
well, the same bug...
=================================
cygcheck: track_down: could not find cygcrypt-0.dll

{ ~ }  » irssi                                                                                  ~
C:/Users/Owl/.babun/cygwin/bin/irssi.exe: error while loading shared libraries: ?: cannot open shared object file: No such file or directory
{ ~ }  »

**Comment by blambin** [comment]
Yes ,i am using win10 and running update.bat fix me too.

**Comment by feilongfl** [comment]
I meet a similar error,when i use make,it show me nothing and return 127,I try to run `update.bat`,but when it finish, i rerun `make -h` command,it show me another error:

> C:/Users/fei_l/.babun/cygwin/bin/make.exe: error while loading shared libraries: ?: cannot open shared object file: No such file or directory
